networkx/algorithms/isomorphism/tests/VF2++/test_vf2pp.py

```diff
@@ -0,0 +1,98 @@
+import networkx as nx
+import random
+from networkx.algorithms.isomorphism.VF2pp import isomorphic_VF2pp
+
```

Comment:
As discussed in the last meeting, I think the best place to start would be to add some extremely simple test cases to define/check basic behavior. For example, what should happen when one (or both) of the input graphs have no nodes?

Some basic tests for this case might look something like:

```suggestion

def test_both_graphs_empty():
    G = nx.Graph()
    H = nx.Graph()
    assert isomorphic_VF2pp(G, H, {}, {})


def test_first_graph_empty():
    G = nx.Graph()
    H = nx.Graph([(0, 1)])
    assert not isomorphic_VF2pp(G, H, {}, {})


def test_second_graph_empty():
    G = nx.Graph([(0, 1)])
    H = nx.Graph()
    assert not isomorphic_VF2pp(G, H, {}, {})
```

These cases might seem extremely basic and even redundant, but simple checks like these form the basis of good, comprehensive unit testing! Note also that I've implicitly made some choices about behavior above; i.e. that two null graphs are isomorphic and any non-null graph is *not* isomorphic to a null graph. Is this correct? Or are some of these (i.e. null to null) pointless concepts? I'm not really sure, but tests like this help frame the discussion for these corner cases and will end up codifying the behavior!

This particular example highlights why writing tests early in the design process can be extremely valuable, as it forces you to think about what the behavior *should* be for common inputs - you can then write the code so that the test cases pass.

Reply:
Thanks a lot for your time and valuable information! I will 100% keep all of what you said in mind. I will make the refinemenet/addition of unit tests a top priority for this week! @rossbar